bin/css-builder/css-builder.js

```diff
@@ -159,13 +159,17 @@ function writeCssDemoFile( theme ) {
  * @param {Object} config
  */
 function writeCssDemoMainFile( theme, config ) {
+	const variableStylesheets = getCssFiles( config.folder, true );
+	variableStylesheets.push( '../../../../espresso-button/style.css' );
```

Comment:
Hmm, okay now I see what you are doing with the button css.  I'm don't  think I like this approach because it "breaks" the modular intent for the css build system where everything in the "root" theme stylesheets is found in the `components/ui/styles` folder and other compiled stylesheets for components etc _enhance_ on top of the root.  I think stylesheets specific to components (in this case buttons) should _enhance_ or simply _use_ what is exposed by the root theme stylesheets because the components stylesheet has a dependency on the root theme stylesheets.

Another problem with your approach here is that the buttons styles are now getting compiled twice.  Once in the root themed stylesheet (currently `CoreAssetsManager::CSS_HANDLE_CORE_CSS_DEFAULT`) and again in the component stylesheet (`CoreAssetManager::CSS_HANDLE_COMPONENTS`).

Reply:
> I think stylesheets specific to components (in this case buttons) should enhance or simply use what is exposed by the root 

I agree with that for the most part, but the issue here is that the styles in the root `buttons.css` file were not targeting a generic html tag like `button`, but were targeting a set of specific css classes like `esprs-button` and `esprs-button-tiny` etc.  Those styles were all being applied by the `EspressoButton` component, and would only appear in the DOM if you either:

- add an `EspressoButton` to your app/component
- copy the css classes applied by `EspressoButton` to some other component/button

I felt that the first option seemed like the most logical and easiest approach, not to mention most DRY.

You have mentioned previously that you preferred to have the css for specific components to only be loaded if that component is being used, so I assumed that you would prefer this approach.

For the sake of clarity I just want to state the obvious point that we should not be adding styles that target generic HTML elements like buttons since that has the potential to override the appearance for elements other than our own. IN some cases this is ok, such as targeting inputs that are a subset of a named form element, like

```css
.ee-form-element input { // styles }
```

anyways, that all said, I assumed you would prefer the `EspressoButton` specific styles to be loaded with that component as opposed to being part of the root css theme. I have no problem with switching that around by:

- moving those button styles back into the `/assets/src/components/ui/styles/root` folder
- removing the stylesheet for the `EspressoButton` component

> the buttons styles are now getting compiled twice

no this is just for pulling in the `EspressoButton` styles for the CSS demo and does NOT add them to the root  theme.